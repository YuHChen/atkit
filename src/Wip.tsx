import React from "react";
import Alert from "react-bootstrap/Alert";

type WipProps = {
  className?: string;
  pageName: string;
};

const Wip = (props: WipProps) => (
  <Alert className={props.className} variant="info">
    <Alert.Heading>Pardon the mess</Alert.Heading>
    <p className="mb-0">This {props.pageName} is still a work in progress.</p>
  </Alert>
);

export default Wip;
