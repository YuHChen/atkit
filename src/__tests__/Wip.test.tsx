import React from "react";
import renderer from "react-test-renderer";

import Wip from "../Wip";

describe("Wip", () => {
  test("renders as expected given minimal props", () => {
    const tree = renderer.create(<Wip pageName="pageName" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  test("renders as expected given full props", () => {
    const tree = renderer
      .create(<Wip className="className" pageName="pageName" />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
