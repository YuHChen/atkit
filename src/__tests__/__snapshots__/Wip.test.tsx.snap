// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wip renders as expected given full props 1`] = `
<div
  className="fade className alert alert-info show"
  role="alert"
>
  <div
    className="alert-heading h4"
  >
    Pardon the mess
  </div>
  <p
    className="mb-0"
  >
    This 
    pageName
     is still a work in progress.
  </p>
</div>
`;

exports[`Wip renders as expected given minimal props 1`] = `
<div
  className="fade alert alert-info show"
  role="alert"
>
  <div
    className="alert-heading h4"
  >
    Pardon the mess
  </div>
  <p
    className="mb-0"
  >
    This 
    pageName
     is still a work in progress.
  </p>
</div>
`;
